koku/masu/processor/parquet/ocp_cloud_parquet_report_processor.py

```diff
@@ -118,15 +127,16 @@ def _determin_s3_path(self, file_type):
     def create_ocp_on_cloud_parquet(self, data_frame, parquet_base_filename, file_number):
         """Create a parquet file for daily aggregated data."""
         # Add the OCP UUID in case multiple clusters are running on this cloud source.
-        if self._provider_type == Provider.PROVIDER_GCP:
+        if self._provider_type in {Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL}:
             if data_frame.first_valid_index() is not None:
-                parquet_base_filename = (
-                    f"{data_frame['invoice_month'].values[0]}{parquet_base_filename[parquet_base_filename.find('_'):]}"
-                )
+                parquet_base_filename = f"{data_frame['invoice_month'].values[0]}_{uuid4()}"
         file_name = f"{parquet_base_filename}_{file_number}_{PARQUET_EXT}"
         file_path = f"{self.local_path}/{file_name}"
         self._write_parquet_to_file(file_path, file_name, data_frame, file_type=self.report_type)
-        self.create_parquet_table(file_path, daily=True)
+        if self.provider_type in {Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL}:
+            self.create_parquet_table(file_path, daily=True, partition_map=GCP_PARTITION_MAP)
+        else:
+            self.create_parquet_table(file_path, daily=True)
```

Comment:
we could make a `partition_map` property:
```
    @property
    def partition_map(self):
        """Post processor based on provider type."""
        if self.provider_type in {Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL}:
            return GCP_PARTITION_MAP
        return None
```

Then this could become:
```suggestion
        self.create_parquet_table(file_path, daily=True, partition_map=self.partition_map)
```

It would just be nice to get away from the `if GCP` specific logic when we are not in a GCP specific class.

Reply:
This was a good move, implemented the suggestion.